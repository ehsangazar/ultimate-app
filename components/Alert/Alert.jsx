/** @jsx jsx */
import { css, jsx } from '@emotion/core'
import { useTheme } from 'emotion-theming'

const Alert = ({ children }) => {
  const theme = useTheme()
  return (
    <div
      css={css`
        text-decoration: none;
        border: none;
        color: ${theme.colors.linkPrimary};
        border-color: ${theme.colors.linkPrimary};
        background-color: ${theme.colors.tertiary};
        padding-top: ${theme.spaces[2]};
        padding-bottom: ${theme.spaces[2]};
        padding-right: ${theme.spaces[4]};
        padding-left: ${theme.spaces[4]};
        color: ${theme.colors.white};
        transition-duration: 0.4s;
      `}
    >
      {children}
    </div>
  )
}

export default Alert
