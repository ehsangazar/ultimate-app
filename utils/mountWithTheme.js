import React from 'react'
import { ThemeProvider } from 'emotion-theming'
import renderer from 'react-test-renderer'
import theme from '../configs/theme'

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  )
}

export default mountWithTheme
