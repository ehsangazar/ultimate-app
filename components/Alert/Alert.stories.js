import React from 'react'
import Alert from './Alert'

export default { title: 'Button' }

export const simple = () => <Alert>Alert</Alert>
