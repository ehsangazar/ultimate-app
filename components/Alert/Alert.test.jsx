import React from 'react'
import serializer from 'jest-emotion'
import mountWithTheme from '../../utils/mountWithTheme'
import Alert from './Alert'

expect.addSnapshotSerializer(serializer)

describe('Alert', () => {
  test('Renders Correctly', () => {
    const component = mountWithTheme(<Alert> Alert</Alert>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
