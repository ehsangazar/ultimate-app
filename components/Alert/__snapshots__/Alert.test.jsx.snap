// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Renders Correctly 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  color: #2c3e50;
  border-color: #2c3e50;
  background-color: #e74c3c;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-right: 16px;
  padding-left: 16px;
  color: #ffffff;
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
}

<div
  className="emotion-0"
>
   Alert
</div>
`;
